ThingIFSDK/ThingIFSDK/QueryClause.swift

```diff
@@ -281,18 +281,16 @@ open class AndClauseInQuery: QueryClause, BaseAnd {
 
      - Parameter clauses: Clause array for And clauses
      */
-    public convenience init(_ clause: QueryClause...) {
+    public init(_ clause: QueryClause...) {
         self.init(clause)
     }
 
     /** Add a clause to And clauses.
 
      - Parameter clause: Clause to be added to and clauses.
      */
-    @discardableResult
-    open func add(_ clause: QueryClause) -> Self {
+    public mutating func add(_ clause: QueryClause) -> Void {
```

Comment:
@yongpingchen It is impossible because `struct` is value type. Even if we return `self` in this method, new struct object is created at calling point.